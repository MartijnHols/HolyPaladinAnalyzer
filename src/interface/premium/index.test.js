import React from 'react';
import { shallow } from 'enzyme';

import { Premium } from './index';

describe('Premium', () => {
  it('matches snapshot when logged out', () => {
    const tree = shallow((
      <Premium />
    ));
    expect(tree).toMatchSnapshot();
  });
  it('matches snapshot when logged in with GitHub', () => {
    const tree = shallow((
      <Premium
        user={{
          name: 'Martijn Hols',
          github: {
            premium: true,
            expires: '2018-07-02T11:48:40.000Z',
          },
        }}
        dateToLocaleString={a => a.toUTCString()}
      />
    ));
    expect(tree).toMatchSnapshot();
  });
  it('matches snapshot when logged in with Patreon', () => {
    const tree = shallow((
      <Premium
        user={{
          name: 'Martijn Hols',
          patreon: {
            premium: true,
          },
        }}
      />
    ));
    expect(tree).toMatchSnapshot();
  });
});
