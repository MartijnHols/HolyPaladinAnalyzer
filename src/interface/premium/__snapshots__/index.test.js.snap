// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Premium matches snapshot when logged in with GitHub 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-3 col-lg-offset-1 col-md-4 col-sm-5"
  >
    <Connect(LoginPanel) />
  </div>
  <div
    className="col-lg-7 col-md-8 col-sm-7"
  >
    <div
      className="panel"
    >
      <div
        className="panel-heading"
      >
        <h2>
          WoWAnalyzer premium
        </h2>
      </div>
      <div
        className="panel-body"
      >
        <Icon
          style={
            Object {
              "color": "#e45a5a",
              "float": "right",
              "fontSize": "6em",
              "marginTop": 0,
            }
          }
        />
        <div
          style={
            Object {
              "fontSize": "1.4em",
              "fontWeight": 400,
            }
          }
        >
          Help out development to unlock 
          <span
            style={
              Object {
                "color": "#e45a5a",
                "fontWeight": 700,
              }
            }
          >
            WoWAnalyzer Premium
          </span>
          !
        </div>
        <br />
        <div
          className="row"
          style={
            Object {
              "marginBottom": 5,
              "marginTop": "2em",
            }
          }
        >
          <div
            className="col-md-12 text-center text-muted"
          >
            How to unlock WoWAnalyzer Premium:
          </div>
        </div>
        <div
          className="row flex"
        >
          <div
            className="col-md-6"
            style={
              Object {
                "borderRight": "1px solid #aaa",
              }
            }
          >
            <h2>
              Patreon
            </h2>
            Help fund further development by becoming a patron on Patreon. Pledge whatever you want!
            <br />
            <br />
            <PatreonButton
              text="Become a Patron"
            />
          </div>
          <div
            className="col-md-6"
          >
            <h2>
              GitHub
            </h2>
            Improve the analysis of a spec or build a new feature to get 1 month of Premium free
            <dfn
              data-tip="Only commits that are merged to the master branch are eligible. Your work will have to pass a pull request review before it can be merged."
            >
              *
            </dfn>
            .
            <br />
            <br />
            <GithubButton
              text="View on GitHub"
            />
          </div>
        </div>
        <div
          className="row"
          style={
            Object {
              "marginBottom": 5,
              "marginTop": "2em",
            }
          }
        >
          <div
            className="col-md-12 text-center text-muted"
          >
            WoWAnalyzer Premium unlocks the following things:
          </div>
        </div>
        <div>
          <div
            className="premium-feature flex"
          >
            <div
              className="content-middle flex-sub"
            >
              <Icon />
            </div>
            <div
              className="flex-main"
            >
              <h2>
                New things
              </h2>
              Your contributions will help fund new things for the site, making it even better.
              <br />
              <br />
              We'll post bounties on the best ideas via 
              <a
                href="https://www.bountysource.com/teams/wowanalyzer"
              >
                Bountysource
              </a>
               as a motivation to get developers to build them.
            </div>
          </div>
          <div
            className="premium-feature flex"
          >
            <div
              className="content-middle flex-sub"
            >
              <Icon />
            </div>
            <div
              className="flex-main"
            >
              <h2>
                No ads
              </h2>
              We'll remove ads from the platform for you so you can consume our content with less distractions and less clutter.
            </div>
          </div>
          <div
            className="premium-feature flex"
          >
            <div
              className="content-middle flex-sub"
            >
              <Icon
                style={
                  Object {
                    "color": "#ff8000",
                  }
                }
              />
            </div>
            <div
              className="flex-main"
            >
              <h2>
                Discord name color
              </h2>
              Get a distinct Discord name color befitting your contribution. See Patreon for Patron specific name colors. Serious GitHub contributors get the yellow contributor name color.
            </div>
          </div>
          <div
            className="premium-feature flex"
          >
            <div
              className="content-middle flex-sub"
            >
              <Icon />
            </div>
            <div
              className="flex-main"
            >
              <h2>
                Access to secret channels on Discord
              </h2>
              Get access to special Discord channels to discuss things privately in the sub-community.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="panel"
    >
      <div
        className="panel-heading"
      >
        <h2>
          You
        </h2>
      </div>
      <div
        className="panel-body"
      >
        Hello 
        Martijn Hols
        . Your Premium is currently 
        <span
          className="text-danger"
        >
          Inactive
        </span>
         
        because of a recent GitHub contribution (active until 
        Mon, 02 Jul 2018 11:48:40 GMT
        )
        . 
        You can get Premium by becoming a Patron on Patreon or by making a contribution to application on GitHub. Try logging in again if you wish to refresh your status.
      </div>
    </div>
  </div>
</div>
`;

exports[`Premium matches snapshot when logged in with Patreon 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-3 col-lg-offset-1 col-md-4 col-sm-5"
  >
    <Connect(LoginPanel) />
  </div>
  <div
    className="col-lg-7 col-md-8 col-sm-7"
  >
    <div
      className="panel"
    >
      <div
        className="panel-heading"
      >
        <h2>
          WoWAnalyzer premium
        </h2>
      </div>
      <div
        className="panel-body"
      >
        <Icon
          style={
            Object {
              "color": "#e45a5a",
              "float": "right",
              "fontSize": "6em",
              "marginTop": 0,
            }
          }
        />
        <div
          style={
            Object {
              "fontSize": "1.4em",
              "fontWeight": 400,
            }
          }
        >
          Help out development to unlock 
          <span
            style={
              Object {
                "color": "#e45a5a",
                "fontWeight": 700,
              }
            }
          >
            WoWAnalyzer Premium
          </span>
          !
        </div>
        <br />
        <div
          className="row"
          style={
            Object {
              "marginBottom": 5,
              "marginTop": "2em",
            }
          }
        >
          <div
            className="col-md-12 text-center text-muted"
          >
            How to unlock WoWAnalyzer Premium:
          </div>
        </div>
        <div
          className="row flex"
        >
          <div
            className="col-md-6"
            style={
              Object {
                "borderRight": "1px solid #aaa",
              }
            }
          >
            <h2>
              Patreon
            </h2>
            Help fund further development by becoming a patron on Patreon. Pledge whatever you want!
            <br />
            <br />
            <PatreonButton
              text="Become a Patron"
            />
          </div>
          <div
            className="col-md-6"
          >
            <h2>
              GitHub
            </h2>
            Improve the analysis of a spec or build a new feature to get 1 month of Premium free
            <dfn
              data-tip="Only commits that are merged to the master branch are eligible. Your work will have to pass a pull request review before it can be merged."
            >
              *
            </dfn>
            .
            <br />
            <br />
            <GithubButton
              text="View on GitHub"
            />
          </div>
        </div>
        <div
          className="row"
          style={
            Object {
              "marginBottom": 5,
              "marginTop": "2em",
            }
          }
        >
          <div
            className="col-md-12 text-center text-muted"
          >
            WoWAnalyzer Premium unlocks the following things:
          </div>
        </div>
        <div>
          <div
            className="premium-feature flex"
          >
            <div
              className="content-middle flex-sub"
            >
              <Icon />
            </div>
            <div
              className="flex-main"
            >
              <h2>
                New things
              </h2>
              Your contributions will help fund new things for the site, making it even better.
              <br />
              <br />
              We'll post bounties on the best ideas via 
              <a
                href="https://www.bountysource.com/teams/wowanalyzer"
              >
                Bountysource
              </a>
               as a motivation to get developers to build them.
            </div>
          </div>
          <div
            className="premium-feature flex"
          >
            <div
              className="content-middle flex-sub"
            >
              <Icon />
            </div>
            <div
              className="flex-main"
            >
              <h2>
                No ads
              </h2>
              We'll remove ads from the platform for you so you can consume our content with less distractions and less clutter.
            </div>
          </div>
          <div
            className="premium-feature flex"
          >
            <div
              className="content-middle flex-sub"
            >
              <Icon
                style={
                  Object {
                    "color": "#ff8000",
                  }
                }
              />
            </div>
            <div
              className="flex-main"
            >
              <h2>
                Discord name color
              </h2>
              Get a distinct Discord name color befitting your contribution. See Patreon for Patron specific name colors. Serious GitHub contributors get the yellow contributor name color.
            </div>
          </div>
          <div
            className="premium-feature flex"
          >
            <div
              className="content-middle flex-sub"
            >
              <Icon />
            </div>
            <div
              className="flex-main"
            >
              <h2>
                Access to secret channels on Discord
              </h2>
              Get access to special Discord channels to discuss things privately in the sub-community.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="panel"
    >
      <div
        className="panel-heading"
      >
        <h2>
          You
        </h2>
      </div>
      <div
        className="panel-body"
      >
        Hello 
        Martijn Hols
        . Your Premium is currently 
        <span
          className="text-danger"
        >
          Inactive
        </span>
         because of your Patreonage
        . 
        You can get Premium by becoming a Patron on Patreon or by making a contribution to application on GitHub. Try logging in again if you wish to refresh your status.
      </div>
    </div>
  </div>
</div>
`;

exports[`Premium matches snapshot when logged out 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-3 col-lg-offset-1 col-md-4 col-sm-5"
  >
    <Connect(LoginPanel) />
  </div>
  <div
    className="col-lg-7 col-md-8 col-sm-7"
  >
    <div
      className="panel"
    >
      <div
        className="panel-heading"
      >
        <h2>
          WoWAnalyzer premium
        </h2>
      </div>
      <div
        className="panel-body"
      >
        <Icon
          style={
            Object {
              "color": "#e45a5a",
              "float": "right",
              "fontSize": "6em",
              "marginTop": 0,
            }
          }
        />
        <div
          style={
            Object {
              "fontSize": "1.4em",
              "fontWeight": 400,
            }
          }
        >
          Help out development to unlock 
          <span
            style={
              Object {
                "color": "#e45a5a",
                "fontWeight": 700,
              }
            }
          >
            WoWAnalyzer Premium
          </span>
          !
        </div>
        <br />
        <div
          className="row"
          style={
            Object {
              "marginBottom": 5,
              "marginTop": "2em",
            }
          }
        >
          <div
            className="col-md-12 text-center text-muted"
          >
            How to unlock WoWAnalyzer Premium:
          </div>
        </div>
        <div
          className="row flex"
        >
          <div
            className="col-md-6"
            style={
              Object {
                "borderRight": "1px solid #aaa",
              }
            }
          >
            <h2>
              Patreon
            </h2>
            Help fund further development by becoming a patron on Patreon. Pledge whatever you want!
            <br />
            <br />
            <PatreonButton
              text="Become a Patron"
            />
          </div>
          <div
            className="col-md-6"
          >
            <h2>
              GitHub
            </h2>
            Improve the analysis of a spec or build a new feature to get 1 month of Premium free
            <dfn
              data-tip="Only commits that are merged to the master branch are eligible. Your work will have to pass a pull request review before it can be merged."
            >
              *
            </dfn>
            .
            <br />
            <br />
            <GithubButton
              text="View on GitHub"
            />
          </div>
        </div>
        <div
          className="row"
          style={
            Object {
              "marginBottom": 5,
              "marginTop": "2em",
            }
          }
        >
          <div
            className="col-md-12 text-center text-muted"
          >
            WoWAnalyzer Premium unlocks the following things:
          </div>
        </div>
        <div>
          <div
            className="premium-feature flex"
          >
            <div
              className="content-middle flex-sub"
            >
              <Icon />
            </div>
            <div
              className="flex-main"
            >
              <h2>
                New things
              </h2>
              Your contributions will help fund new things for the site, making it even better.
              <br />
              <br />
              We'll post bounties on the best ideas via 
              <a
                href="https://www.bountysource.com/teams/wowanalyzer"
              >
                Bountysource
              </a>
               as a motivation to get developers to build them.
            </div>
          </div>
          <div
            className="premium-feature flex"
          >
            <div
              className="content-middle flex-sub"
            >
              <Icon />
            </div>
            <div
              className="flex-main"
            >
              <h2>
                No ads
              </h2>
              We'll remove ads from the platform for you so you can consume our content with less distractions and less clutter.
            </div>
          </div>
          <div
            className="premium-feature flex"
          >
            <div
              className="content-middle flex-sub"
            >
              <Icon
                style={
                  Object {
                    "color": "#ff8000",
                  }
                }
              />
            </div>
            <div
              className="flex-main"
            >
              <h2>
                Discord name color
              </h2>
              Get a distinct Discord name color befitting your contribution. See Patreon for Patron specific name colors. Serious GitHub contributors get the yellow contributor name color.
            </div>
          </div>
          <div
            className="premium-feature flex"
          >
            <div
              className="content-middle flex-sub"
            >
              <Icon />
            </div>
            <div
              className="flex-main"
            >
              <h2>
                Access to secret channels on Discord
              </h2>
              Get access to special Discord channels to discuss things privately in the sub-community.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
